redpanda/templates/configmap.yaml

```diff
@@ -39,30 +54,35 @@ data:
     organization: {{ .Values.organization }}
     {{- end }}
     redpanda:
-{{- if .Values.auth.sasl.enabled }}
-  {{- with .Values.auth.sasl.users }}
+{{- if not (include "redpanda.semver" . | semverCompare ">=22.1.4") }}
+  {{- if .Values.auth.sasl.enabled }}
+    {{- with .Values.auth.sasl.users }}
       superusers:
     {{ toYaml . | indent 8 }}
+    {{- end }}
+  {{- end }}
+  {{- if not (eq (toYaml .Values.config) "null") }}
```

Comment:
This isn't a request to change, but is this the same as the following line?
```
{{- if not (hasKey .Values.listeners.rpc "tls") }}
```

Reply:
After testing I see that these sections appear as `key: null` in the configmap output (rather than not existing). I think this is also related to a bug we discussed internally with using an older Redpanda version, where the cluster won't start with this branch.